fix(core.configuration): write snapshots with 600 permissions

MMaiero opened the pull request:
Set snapshot file permissions to owner read/write only (600) instead of the default permissions (644) to enhance security. Configuration snapshots may contain sensitive data and should not be readable by group or other users.

The change:
- Adds POSIX file permissions support to writeSnapshot method
- Sets permissions to rw------- after file creation
- Gracefully handles non-POSIX filesystems with warning log

MMaiero committed:
fix(core.configuration): write snapshots with 600 permissions

Set snapshot file permissions to owner read/write only (600) instead of
the default permissions (644) to enhance security. Configuration
snapshots may contain sensitive data and should not be readable by
group or other users.

The change:
- Adds POSIX file permissions support to writeSnapshot method
- Sets permissions to rw------- after file creation
- Gracefully handles non-POSIX filesystems with warning log

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero committed:
Updated tests, improved code

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

nicolatimeus commented on kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java:
```diff
@@ -1032,21 +1057,17 @@ private void writeSnapshot(long sid, XmlComponentConfigurations conf) throws Kur
-            }
-            throw new KuraIOException(e);
+            tempSnapshotFile = File.createTempFile(fSnapshot.getName(), null, new File(fSnapshot.getParent()));
+            tempSnapshotFile.deleteOnExit();
```
Is this really necessary? it will add a new exit hook for each snapshot leaking memory. Is it better to delete the temporary file if it exists in a finally block instead?

MMaiero replied:
I think we always had it: https://github.com/eclipse-kura/kura/blob/5c2f432e10e1feef54ec37c134de08094f24951d/kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java#L1014
I have just moved it around to refactor in separate methods to improve readability.

I'll have a look based on your suggestion

MMaiero replied:
actually it was contributed not long ago by @sfiorani https://github.com/eclipse-kura/kura/commit/4f1bcc54bf3fc3926759348b09ab57ae088d94fe

MMaiero replied:
Made a change

MMaiero committed:
fix(core.configuration): improve temporary snapshot file cleanup

- Add proper cleanup of temporary files in finally block
- Remove deleteOnExit() call and implement explicit cleanup
- Add warning logging when cleanup fails
- Ensures temporary files are cleaned up even if snapshot write fails

nicolatimeus commented on kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java:
```diff
@@ -1009,14 +1011,46 @@ private void writeSnapshot(long sid, XmlComponentConfigurations conf) throws Kur
+            logger.warn("Unable to set POSIX file permissions for snapshot file: {}", fSnapshot.getAbsolutePath(), e1);
+        } catch (IOException e) {
+            try {
+                Files.delete(tempSnapshotFile.toPath());
```
I think you can remove this after the last change

MMaiero replied:
Right!

MMaiero committed:
Cleanup as suggested

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>
